test/unit_test_input_parser.c

```diff
@@ -1,6 +1,166 @@
 #include "unit_test_input_parser.h"
 
+// WARNING !!! This test may seg fault if it fails!
```

Comment:
I don't think there is a way to prevent yourself from seg faulting when you don't know how big the array is. I have no way of knowing how much data was actually allocated for the array. So, in order to pass the test, I iterate over the expected number of entries in the array.